src/black/__init__.py

```diff
@@ -6380,11 +6380,18 @@ def diff(a: str, b: str, a_name: str, b_name: str) -> str:
     """Return a unified diff string between strings `a` and `b`."""
     import difflib
 
-    a_lines = [line + "\n" for line in a.splitlines()]
-    b_lines = [line + "\n" for line in b.splitlines()]
-    return "".join(
-        difflib.unified_diff(a_lines, b_lines, fromfile=a_name, tofile=b_name, n=5)
-    )
+    a_lines = [line for line in a.splitlines(keepends=True)]
+    b_lines = [line for line in b.splitlines(keepends=True)]
+    diff_lines = []
+    for line in difflib.unified_diff(
+        a_lines, b_lines, fromfile=a_name, tofile=b_name, n=5
+    ):
+        if line[-1] == "\n":
+            diff_lines.append(line)
+        else:
+            diff_lines.append(line + "\n")
+            diff_lines.append("\\ No newline at end of file\n")
+    return "".join(diff_lines)
```

Comment:
To someone like me, the `diff_lines.append("\\ No newline at end of file\n")` link just looks plain wrong. I would suggest explaining that this code is using a special feature of the patch specification(?) to work around a Python bug. Also linking the BPO would be the cherry on the cake!

Reply:
Yeah, sounds good. Now that I have a BPO reference, I can do that.

Reply:
Done